src/emeus-vfl-parser.c

```diff
@@ -343,6 +359,38 @@ parse_predicate (VflParser *parser,
       return false;
     }
 
+parse_attribute:
+  if (*end == '.')
+    {
+      const char *attr;
+      int i = 0;
+
+      end += 1;
+      attr = valid_attributes[0];
+      while (attr != NULL)
```

Comment:
We could even store the length inside the `valid_attributes` array, e.g.

```
static const struct {
  int len;
  const char *attr;
} valid_attributes[] = {
  { 5, "width" },
  { 6, "height" },
  …
};
```

and then do:

```
for (int i = 0; i < G_N_ELEMENTS (valid_attributes); i++)
  {
    if (g_ascii_strncasecmp (valid_attributes[i].name, end, valid_attributes[i].len) == 0)
      {
        …
      }
  }
```